src/engine/strat_engine/crypt/shared.rs

```diff
@@ -204,14 +204,40 @@ pub fn clevis_info_from_metadata(
     pin_dispatch(&subjson, CLEVIS_RECURSION_LIMIT).map(Some)
 }
 
-/// Returns true if the Tang config has a thumbprint or an advertisement
-/// or all Tang configs in the nested sss config have thumbprints or
-/// advertisements.
-fn all_tang_configs_have_url_trust_info(
+/// Generic traversal of a Clevis config.
+/// Apply the specified simple functions to tang and tpm2 configs.
+/// The sss parameter is the initial value and function for a try_fold
+/// action for an sss pin. The sss function is a FnMut in order to be
+/// able to update the accumulator.
+///
+/// fn example(
+///    pin: &str,
+///    clevis_config: &Value,
+///    recursion_limit: u64,
+/// ) -> StratisResult<Vec<u32>> {
+///    traverse_clevis_config(
+///        pin,
+///        clevis_config,
+///        recursion_limit,
+///        &|_obj| Ok(Vec::new()),
+///        &|_| Ok(Vec::new()),
+///        &mut (&Vec::new(), &mut |mut acc, _premise| {
+///            acc.push(2);
+///            Ok(acc)
+///        }),
+///    )
+/// }
+fn traverse_clevis_config<T>(
     pin: &str,
     clevis_config: &Value,
     recursion_limit: u64,
-) -> StratisResult<bool> {
+    tang_func: &dyn Fn(&Map<String, Value>) -> StratisResult<T>,
+    tpm2_func: &dyn Fn(&Map<String, Value>) -> StratisResult<T>,
+    sss: &mut (&T, &mut dyn FnMut(T, T) -> StratisResult<T>),
```

Comment:
Can you explain why you made these references? I don't see that pattern used in most closure usage in the Rust code I've seen. My understanding is that `Fn()` itself already represents a reference.